### DISCUSSION TITLE
Cannot find options for Generate Visual Studio Project Files in order to connect Unreal environment with Airsim

### DISCUSSION THREAD

**Discussion by Deepakgthomas** [microsoft/AirSim#3248] [Support Q&A] [answered]
I developed an environment using Unreal 4.25. In order to connect it with Airsim, I followed the instructions given over here - https://microsoft.github.io/AirSim/unreal_custenv/.

I'm stuck at the point where they say to right click the .uproject file and select Generate Visual Studio Project Files, since I do not have that option.

![image](https://user-images.githubusercontent.com/21986394/103163588-406dd000-47c5-11eb-97ef-f1ef91178b9a.png)

**Comment by xxEoD2242** [comment]
This happens when installing Unreal Engine. You can do one of two things:

1. Restart your computer. Try to right click again so the tool chain will be installed. I’ve had mixed success.
2. Open the Unreal Engine project. It'll tell you it has missing modules and you need to recompile. Hit yes and it will compile AirSim.

An alternative option is to open the Unreal Project folder (example would be LandscapeMountains) and then run the debug version. This will compile AirSim using Visual Studio, which will display errors if there are any issues.

**Comment by Deepakgthomas** [comment, marked as answer]
Thank you for the advice! However, that was not my issue. I did not restart Epic games after installing Unreal Engine therefore the project file extensions were not associated with my Engine. Please refer to this post for more details - https://github.com/microsoft/AirSim/issues/1219
